theano/tensor/c_code/alt_blas_template.c

```diff
@@ -66,25 +68,27 @@ PyObject* alt_op_%(float_type)s(int to_transpose, %(float_type)s* M, int nrow, i
         strides[0] = %(float_size)d;
         strides[1] = LDM * %(float_size)d;
     }
-    return PyArray_New(&PyArray_Type, 2, dims, %(npy_float)s, strides, M, 0, 0, NULL);
+    return PyArray_New(&PyArray_Type, 2, dims, %(npy_float)s, strides, M, 0, numpyFlags, NULL);
 }
-/* Special wrapping case used for matrix C in gemm implementation. */
+
+/* Special wrapping case used for matrix C in gemm_ implementation. */
 inline PyObject* alt_wrap_fortran_writeable_matrix_%(float_type)s(
     %(float_type)s* matrix, const int* nrow, const int* ncol, const int* LD
 ) {
     npy_intp dims[2] = {*nrow, *ncol};
     npy_intp strides[2] = {%(float_size)d, (*LD) * %(float_size)d};
     return PyArray_New(&PyArray_Type, 2, dims, %(npy_float)s, strides, matrix, 0, NPY_ARRAY_WRITEABLE, NULL);
 }
-/* %(name)s template code */
-void %(name)s(
+
+/* gemm_ template code */
+void %(precision)sgemm_(
```

Comment:
I think you want to remove the s in the name if you add precission here.

Reply:
In fact, the `s` is part of `%(precision)s`, so it is required to format the string :) !

Reply:
right!